packages/snaps-sdk/src/jsx/validation.ts

```diff
@@ -243,7 +243,9 @@ export const ButtonStruct: Describe<ButtonElement> = element('Button', {
   children: children([StringElementStruct, ImageStruct, IconStruct]),
   name: optional(string()),
   type: optional(nullUnion([literal('button'), literal('submit')])),
-  variant: optional(nullUnion([literal('primary'), literal('destructive')])),
+  variant: optional(
+    nullUnion([literal('primary'), literal('destructive'), literal('loading')]),
+  ),
   disabled: optional(boolean()),
```

Comment:
Hmm, should loading be a prop similar to `disabled` instead of a variant?

WDYT @Mrtenz?

Reply:
I think it should, as we'd want to style it differently for `primary` and `destructive` buttons potentially.

Reply:
Even if we wouldn't right now, it makes it easy to do it in the future

Reply:
Ye, I think all states (primary, disabled, destructive) should be able to load, initially with same spinner (inheriting the color of the state)